crypto-primitives/src/merkle_tree/mod.rs

```diff
@@ -255,23 +255,27 @@ impl std::fmt::Display for MerkleTreeError {
     }
 }
 
+#[cfg(feature = "std")]
 impl std::error::Error for MerkleTreeError {
     #[inline]
     fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
         None
     }
 }
 
+#[cfg(not(feature = "std"))]
+impl algebra::Error for MerkleTreeError {}
+
 /// Returns the log2 value of the given number.
 #[inline]
 fn log2(number: usize) -> usize {
-    (number as f64).log2() as usize
+    algebra::log2(number) as usize
 }
 
 /// Returns the height of the tree, given the size of the tree.
 #[inline]
 fn tree_height(tree_size: usize) -> usize {
-    log2(tree_size + 1)
+    log2(tree_size)
```

Comment:
`algebra::log2` returns approximate value of `log2`. To make tests pass, I had to remove `+1` from here. I am not sure whether this changes the logic significantly

Reply:
Maybe we can use this: https://users.rust-lang.org/t/logarithm-of-integers/8506/5

Reply:
Nvm I should have read the code before commenting 🤦🏾‍♂️